core/src/main/scala/cats/data/Continuation.scala

```diff
@@ -0,0 +1,108 @@
+package cats
+package data
+
+import java.io.Serializable
+import scala.annotation.tailrec
+
+sealed abstract class Continuation[O, +I] extends Serializable {
```

Comment:
Assuming my algebra is right I think `O` could also be made covariant like you said. A separate but related question is if we want this to be variant at all since that comes with all the gotchas associated with variance and subtyping in Scala. I would vote no.

I'm guessing the order is because it makes partial application look a bit cleaner since the `Monad` instance is free in `I` as opposed to `R`. with `Continuation[I, O]` you would have `Monad[Continuation[?, O]]` as opposed to `Monad[Continuation[O, ?]]`.